eemeter/modeling/models/billing_caltrack.py

```diff
@@ -0,0 +1,308 @@
+        #     confidence-interval-of-rmse
+        #
+        # > Let xi be your true value for the ith data point and xhat_i the
+        # >   estimated value.
+        # > If we assume that the differences between the estimated and
+        # > true values have
+        # >
+        # > 1. mean zero (i.e. the xhat_i are distributed around xi)
+        # > 2. follow a Normal distribution
+        # > 3. and all have the same standard deviation sigma
+        # > then you really want a confidence interval for sigma
+        # > ...
+        #
+        # We might decide these assumptions don't hold.
+
+        n = self.estimated.shape[0]
+
+        c1, c2 = chi2.ppf([0.025, 1-0.025], n)
+        self.lower = np.sqrt(n/c2) * self.rmse
+        self.upper = np.sqrt(n/c1) * self.rmse
+        self.n = n
+
+        self.plot()
+
+        self.params = {
+            "coefficients": self.result.params,
+            "intercept": self.result.params['Intercept'],
+            "X_design_info": X,
+            "formula": self.formula,
+        }
+
+        output = {
+            "r2": self.r2,
+            "model_params": self.params,
+            "rmse": self.rmse,
+            "cvrmse": self.cvrmse,
+            "upper": self.upper,
+            "lower": self.lower,
+            "n": self.n
+        }
+        return output
+
+    def predict(self, demand_fixture_data, params=None):
+        ''' Predicts across index using fitted model params
+
+        Parameters
+        ----------
+        demand_fixture_data : pandas.DataFrame
+            Formatted input data as returned by
+            :code:`ModelDataBillingFormatter.create_demand_fixture()`
+        params : dict, default None
+            Parameters found during model fit. If None, `.fit()` must be called
+            before this method can be used.
+
+              - :code:`X_design_matrix`: patsy design matrix used in
+                formatting design matrix.
+              - :code:`formula`: patsy formula used in creating design matrix.
+              - :code:`coefficients`: ElasticNetCV coefficients.
+              - :code:`intercept`: ElasticNetCV intercept.
+
+        Returns
+        -------
+        output : pandas.DataFrame
+            Dataframe of energy values as given by the fitted model across the
+            index given in :code:`demand_fixture_data`.
+        '''
+        # needs only tempF
+        if params is None:
+            params = self.params
+
+        data = demand_fixture_data.resample(
+            pd.tseries.frequencies.Day()).agg({'tempF': np.mean})
+        model_data = self._make_monthly(data)
+
+        predicted = self.result.predict(model_data)
+        predicted = pd.Series(predicted, index=model_data.index)
+        for i in range(len(predicted.index)):
+            predicted[i] = predicted[i]* calendar.monthrange(predicted.index[i].year,predicted.index[i].month)[1]
+
+        return predicted
+
+    def plot(self):
```

Comment:
Does this still work? If it's general enough maybe we should factor it out into a mixin or something so that all of the models can use the same code.